app/adapters/alma_adapter.rb

```diff
@@ -56,11 +55,8 @@ def get_catalog_date_from_record(record)
     ava = record.select { |f| f.tag == "AVA" }
     if ava.count > 0
       # Get the creation date from the physical items
-      dates = []
-      get_items_for_bib(record.bib.id).each do |_location, holdings|
-        items = holdings.map { |h| h["items"] }.compact.flatten
-        dates += items.map { |i| i["creation_date"] }.compact.flatten
-      end
+      item_set = get_items_for_bib(record.bib.id)
+      dates = item_set.items.map { |i| i["item_data"]["creation_date"] }
```

Comment:
I am not sure the `creation_date` will always be there, maybe re-adding the `.compact` would be a good idea to prevent `nil` values into the `dates` array.